src/index.rs

```diff
@@ -203,6 +215,10 @@ impl IndexTable {
 		}))
 	}
 
+	// [Review] lazy creation, I don't think it is worth mentioning, pretty obvious
+	// by reading the type of map.
+	// Note that having a directory abstraction could ensure two identical path
+	// at not use at small cost.
```

Comment:
Not sure what you mean here. `id` should result in a unique filename, so there should be no path collisions.

Reply:
Yes I don't remember very well, was probably thinking about a db directory abstraction, but it is more an idea for code organisation. No conflict with current code can happen indeed.